optimizely/decision/experiment_whitelist_service.go

```diff
@@ -0,0 +1,74 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package decision //
+package decision
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/optimizely/go-sdk/optimizely/decision/reasons"
+
+	"github.com/optimizely/go-sdk/optimizely/entities"
+)
+
+// ExperimentWhitelistService makes a decision using a whitelist (a set of experiment + variation assignments for a set of users)
+// whitelist should be a map of user ID, to a map of Experiment key to Variation key
+type ExperimentWhitelistService struct {
+	whitelist map[string]map[string]string
+}
+
+// NewExperimentWhitelistService returns a new instance of ExperimentWhitelistService
+func NewExperimentWhitelistService(whitelist map[string]map[string]string) *ExperimentWhitelistService {
```

Comment:
who/what will create whitelist map ?

Reply:
We could have the factory option accept this map and pass it to this service. But do you think there's a better/more convenient format for the user use to specify the whitelist?